.drone.yml

```diff
@@ -0,0 +1,96 @@
+branches:
+  - dev
+  - master
+  - release-*
+  - testing
+
+__buildenv: &buildenv
+  image: rchain/buildenv:latest-java8
+
+__sbtenv: &sbtenv
+  <<: *buildenv
+  environment:
+    - _JAVA_OPTIONS=-Xms2G -Xmx4G -XX:MaxMetaspaceSize=1G
+  volumes:
+    - /var/cache/rchain-build/.sbt:/root/.sbt
+    - /var/cache/rchain-build/.ivy2:/root/.ivy2
+
+clone:
+  git-clone:
+    image: plugins/git:next
+
+pipeline:
+
+  # pr
+
+  sbt-update:
+    <<: *sbtenv
+    commands:
+      - sbt update
+
+  compile:
+    <<: *sbtenv
+    commands:
+      - sbt rholang/bnfc:generate compile test:compile it:compile
+
+  run-unit-tests:
+    <<: *sbtenv
+    commands:
+      - sbt test it:test
+
+  run-code-analysis:
+    <<: *sbtenv
+    commands:
+      - sbt scalafmtCheck coverage
+      - ./scripts/ci/codecov-upload
+
+  # push
+
+  package:
+    <<: *sbtenv
+    commands:
+      - sbt node/docker:publishLocal node/rpm:packageBin node/debian:packageBin node/universal:packageZipTarball
+    volumes:
+      - /var/cache/rchain-build/.sbt:/root/.sbt
+      - /var/cache/rchain-build/.ivy2:/root/.ivy2
+      - /var/run/docker.sock:/var/run/docker.sock
+    when:
+      event: push
```

Comment:
We should be doing this for the PR build as well, otherwise we don't know we've broken packaging till we're in dev.

Reply:
I see, the docker sock thing is very unfortunate.

I think though since the build definition is taken from the branch being built, the drone machine can be compromised by **any** PR, regardless of what we put in .drone.yml. I can put whatever I want in the .drone.yml in my PR.

Unless we're using gated/protected builds:
https://discourse.drone.io/t/gated-builds-example-usage-help/702/4
Then every change in .drone.yml needs approval before being run.

Gated builds are underdocumented, but we have to enable those (or disable PR builds whatsoever ;/)

We should make a JIRA issue for that.

---

I'm reluctant to agree that checking packaging logic is time wasted for PR builds, but admittedly this fails much less often than integration tests do.

> if someone breaks it, merge-push build will catch it, just as it catches other regressions in integration test.

That's way too late. See the poll result.

I think we could guard the parts using docker-sock by checking the PR/branch author is in our ORG (we could call github or use a secret with all allowed usernames). Since we have to `guard` the .drone.yml anyway, putting the check into the .yml should make it safe.

---

Also, on the docker.sock / security issue:
Is the drone docker host in a DMZ subnet?
Is it running in a VM?
I think - given the threats - both answers should be true.

Reply:
BTW: how was this docker.sock issue solved for gitlab? I'm sure we're building docker images there - what prevents a PR/branch author from getting root in there?